java/test.sh

```diff
@@ -2,36 +2,31 @@
 
 # Cause the script to exit if a single command fails.
 set -e
-
 # Show explicitly which commands are currently running.
 set -x
 
 ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
-$ROOT_DIR/../build.sh -l java
+
+# Skip compiling the binaries if `--skip-compile` is passed in.
+if [[ "$@" != *--skip-compile* ]]; then
+    echo "Compiling binaries."
+    $ROOT_DIR/../build.sh -l python,java
```

Comment:
This line seems not to install python ray (just build) and `which ray` may still be missing.

Reply:
I added `./ci/travis/install-ray.sh` in the java job, which will install ray. However these 2 commands will do `build.sh` twice. I'll merge them in the next PR.

Reply:
nvm, fixed in this PR.